lib/module.js

```diff
@@ -3,47 +3,31 @@ const { promisify } = require('util')
 const { removeSync, outputFile } = require('fs-extra')
 const { Feed } = require('feed')
 const AsyncCache = require('async-cache')
-const consola = require('consola')
-
-const logger = consola.withScope('nuxt:feed')
+const logger = require('./logger')
 
 const defaults = {
   path: '/feed.xml',
   async create(feed) {},
   cacheTime: 1000 * 60 * 15
 }
 
-async function feedModule() {
-  // Factory function
-  if (typeof this.options.feed === 'function') {
-    this.options.feed = await this.options.feed()
-  }
-
-  // Factory object
-  if (!Array.isArray(this.options.feed)) {
-    if (this.options.feed.factory) {
-      this.options.feed = await this.options.feed.factory(this.options.feed.data)
-    }
-  }
-
-  // Single feed
-  if (!Array.isArray(this.options.feed)) {
-    this.options.feed = [this.options.feed]
-  }
-
-  const options = Object.assign([], this.options.feed).map(o => Object.assign({}, defaults, o))
+async function feedModule(moduleOptions) {
+  const options = [
+    ...await parseOptions(moduleOptions),
```

Comment:
I wouldn't merge the options. Instead, check whether moduleOptions are populated. If so, use them. Else, use `this.options.feed`

Reply:
I do not know if I agree, most modules mix, today is not even being used

Reply:
@manniL ao you think we should concat instead of merge?

Reply:
@ricardogobbosouza it doesn't make sense for the user to use both options. :man_shrugging:

Reply:
That's why I removed the moduleOptions support for now.

I don't see a reason why we should add that for this module.

Reply:
@manniL 
Actually, the two together does not make sense, but so we can use or not.
I see no reason to change here

Reply:
Follow the same module https://github.com/nuxt-community/redirect-module/blob/master/lib/module.js#L2